lib/scoreboard.rb

```diff
@@ -0,0 +1,92 @@
+class Scoreboard
+  def initialize(player_name, io)
+    @player = Player.new(player_name)
+    @scorecard = []
+    @frame_number = 0
+    @io = io
+  end
+
+  def player
+    return @player
+  end
+
+  def scorecard
+    return @scorecard
+  end
+
+  # ONE FRAME OF A PLAYER'S GAME
+  def play_frame
+    frame = Frame.new(@player)
+    @io.puts 'roll one'
+    @io.puts 'enter the number of pins knocked down'
+    x = @io.gets.chomp.to_i
+    frame.roll_one(@player.roll(x))
+
+    if frame.stats[:special] != 'STRIKE'
+      @io.puts 'roll two'
+      @io.puts 'enter the number of pins knocked down'
+      y = @io.gets.chomp.to_i
+      frame.roll_two(@player.roll(y))
+    end
+
+    @scorecard << frame.stats
+
+    # SPARE AND STRIKE CONDITIONS
+    if @scorecard[@scorecard.size-2][:special] == 'SPARE'
+      @scorecard[@scorecard.size-2][:score] += x
+    end
+
+    if @scorecard[@scorecard.size-2][:special] == 'STRIKE' && frame.stats[:special] == 'none'
+      @scorecard[@scorecard.size-2][:score] += x
+      @scorecard[@scorecard.size-2][:score] += y
+    end
+
+    if @scorecard[@scorecard.size-2][:special] == 'STRIKE' && frame.stats[:special] == 'STRIKE'
+      @scorecard[@scorecard.size-2][:score] += x
+    end
+
+    if @scorecard.size > 3
+      if @scorecard[@scorecard.size-2][:special] == 'STRIKE' && @scorecard[@scorecard.size-3][:special] == 'STRIKE'
+        @scorecard[@scorecard.size-3][:score] += x
+      end
+    end
+
```

Comment:
If statements are not too complicated so it makes the code easy to follow and read.